wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/jpa/JpaUserDevfileDao.java

```diff
@@ -0,0 +1,420 @@
+    }
+
+    public static UserDevfileSearchQueryBuilder newBuilder(EntityManager entityManager) {
+      return new JpaUserDevfileDao.UserDevfileSearchQueryBuilder(entityManager);
+    }
+
+    public UserDevfileSearchQueryBuilder withMaxItems(int maxItems) {
+      this.maxItems = maxItems;
+      return this;
+    }
+
+    public UserDevfileSearchQueryBuilder withSkipCount(int skipCount) {
+      this.skipCount = skipCount;
+      return this;
+    }
+
+    public UserDevfileSearchQueryBuilder withFilter(List<Pair<String, String>> filter) {
+      if (filter == null || filter.isEmpty()) {
+        this.filter = "";
+        return this;
+      }
+      final StringJoiner matcher = new StringJoiner(" AND ", " WHERE ", " ");
+      int i = 0;
+      for (Pair<String, String> attribute : filter) {
+
+        final String parameterName = "parameterName" + i++;
+        if (attribute.second.startsWith("like:")) {
+          params.put(parameterName, attribute.second.substring(5));
+          matcher.add("userdevfile." + attribute.first + " LIKE :" + parameterName);
+        } else {
+          params.put(parameterName, attribute.second);
+          matcher.add("userdevfile." + attribute.first + " = :" + parameterName);
+        }
+      }
+      this.filter = matcher.toString();
+      return this;
+    }
+
+    public UserDevfileSearchQueryBuilder withOrder(List<Pair<String, String>> order) {
+      if (order == null || order.isEmpty()) {
+        this.order = "";
+        return this;
+      }
+      final StringJoiner matcher = new StringJoiner(", ", " ORDER BY ", " ");
+      order.forEach(pair -> matcher.add("userdevfile." + pair.first + " " + pair.second));
+      this.order = matcher.toString();
+
+      return this;
+    }
+
+    public TypedQuery<Long> buildCountQuery() {
+      StringBuilder query =
+          new StringBuilder()
+              .append("SELECT ")
+              .append(" COUNT(userdevfile) ")
+              .append("FROM UserDevfile userdevfile")
+              .append(filter);
+      TypedQuery<Long> typedQuery = entityManager.createQuery(query.toString(), Long.class);
+      params.forEach((k, v) -> typedQuery.setParameter(k, v));
+      return typedQuery;
+    }
+
+    public TypedQuery<UserDevfileImpl> buildSelectItemsQuery() {
+
+      StringBuilder query =
+          new StringBuilder()
+              .append("SELECT ")
+              .append(" userdevfile ")
+              .append("FROM UserDevfile userdevfile")
+              .append(filter)
+              .append(order);
+      TypedQuery<UserDevfileImpl> typedQuery =
+          entityManager
+              .createQuery(query.toString(), UserDevfileImpl.class)
+              .setFirstResult(skipCount)
+              .setMaxResults(maxItems);
+      params.forEach((k, v) -> typedQuery.setParameter(k, v));
+      return typedQuery;
+    }
+  }
+
+  @Singleton
+  public static class RemoveUserDevfileBeforeAccountRemovedEventSubscriber
```

Comment:
This is technically indenpendent of the Jpa dao. IMHO it would be better placed as a top level class instead of an inner class of the dao.